### ISSUE TITLE
Why use CUDA Extention here?

### ISSUE THREAD

**Issue by minygd** [feihuzhang/GANet#63]
Hello, sorry to bother you : ) 
I have a question about the effectiveness of the CUDA Extention, beacuse I never see Deep Learning models like this before. Is this really necessary in your `libs` file?

**Comment by feihuzhang** [comment]
Hope you could know that all Pytorch or tensorflow functions are actually built on CUDA and C++ similar to this. Python has very low efficiency. Auto-gradient is not a panacea.

**Comment by minygd** [comment]
OK, THX！
